benchopt/base.py

```diff
@@ -592,3 +594,96 @@ def repeat():
         cv_fold = next(self._cv)
         split_ = getattr(self, "split", self._default_split)
         return split_(cv_fold, *arrays)
+
+
+class BasePlot(ParametrizedNameMixin, DependenciesMixin, ABC):
+    _base_class_name = 'Plot'
+
+    @abstractmethod
+    def plot(self, df, **kwargs):
+        ...
+
+    def get_name(self):
+        return self.name.replace(" ", "_")
+
+    def check(self):
+        self.check_type()
+        self.check_params()
+
+    def check_type(self):
+        if not hasattr(self, 'type'):
+            raise ValueError("Plot should have a `type` attribute.")
+        supported_types = ['scatter']
+        if self.type not in supported_types:
+            raise ValueError(
+                f"Plot type should be one of {' '.join(supported_types)}. "
+                f"Got {self.type}."
+            )
+
+    def check_params(self):
```

Comment:
I think using `dropdown` would be more explicit here, as this is what is gonna be used to create dropdown?
```suggestion
    def check_dropdown(self):
```